app/operations/story_operations/update.rb

```diff
@@ -0,0 +1,96 @@
+module StoryOperations
+  class Update
+    include Operation
+
+    def initialize(story:, data:, current_user:)
+      @story = story
+      @data = data
+      @current_user = current_user
+    end
+
+    def call
+      ActiveRecord::Base.transaction do
+        yield ensure_valid_state
+        yield documents_attributes_changes
+        yield update_story
+
+        yield create_changesets
+        yield apply_fixes
+
+        yield notify_state_changed
+        yield notify_users
+        yield notify_changes
+
+        yield create_activity
+
+        Success(story)
+      end
+    rescue
+      Failure(story)
+    end
+
+    private
+
+    attr_reader :story, :data, :current_user
+
+    def should_be_unscheduled?(estimate:, type:)
+      Story.can_be_estimated?(type) && estimate.blank?
+    end
+
+    def ensure_valid_state
+      return Success(data) unless should_be_unscheduled?(
+        estimate: data[:estimate],
+        type: data[:story_type]
+      )
+
+      data[:state] = 'unscheduled'
+      Success(data)
```

Comment:
Since both cases return the `data`, what do you think about:

```suggestion
      if should_be_unscheduled?(
        estimate: data[:estimate],
        type: data[:story_type]
      )
        data[:state] = 'unscheduled'
      end
      
      Success(data)
```

Reply:
much better